### ISSUE TITLE
Monitor specific images(whether container is running or not)

### ISSUE THREAD

**Issue by kramerrs** [containrrr/watchtower#1597]
### Is your feature request related to a problem? Please describe.

 I am using a service ShinyProxy.  ShinyProxy manages launching of containers.  The containers I want updating may not be running.  ShinyProxy also launches services with randomized container names.  Even if they didn't randomize the container names, it would continuously launch new containers with new names, because ShinyProxy launches the containers as docker services.  EG the users are isolated in their own container.  One container per user per application.

### Describe the solution you'd like

The desired solution is to specify the images including registry and tags for changes, then pull/retag relaunch any images that have changed.  

### Describe alternatives you've considered

Reverse engineer the authentication code and set up a webhook through Portainer.

**Comment by kramerrs** [comment]
Seems like the functionality is almost there already.  When I look at the code there is already, a FilterByImage function right by the FilterByName, and the BuildFilter function.

I would say for a the ShinyProxy container, searching by regex of the names would also work, since it prefixes the randomized name.

I might take a crack at modding the code.

**Comment by kramerrs** [comment]
I found the regex easter egg, in the container names.

I am going to go ahead and close the issue, as that appears good enough for me.  However it would be nice if I could just specify some images, so I don't have to launch the container, to get watchtower to update.  It only happens the first time I push an update, so not a big issue, just kind of weird that it wouldn't check at the image level.  I see the code for it already in the pkg.

**Comment by piksel** [comment]
I am not sure what you mean by `running` and `launching`, but you can check stopped containers as well. If there is no container at all, then there wouldn't be anything for watchtower to recreate... 

The fact that you can use regex to match container names should be in the docs though, but it seems like it was omitted.

**Comment by kramerrs** [comment]
> I am not sure what you mean by `running` and `launching`, but you can check stopped containers as well. If there is no container at all, then there wouldn't be anything for watchtower to recreate...
> 
> The fact that you can use regex to match container names should be in the docs though, but it seems like it was omitted.

Yeah, I think I am using Watchtower in lieu of the docker pull policy being convoluted.  EG, I really just want docker to pull and retag images before they are launched, however docker pull policy "always", pulls the image but if there is an image with the same tag in the local repository, the new image is left without a tag, so ShinyProxy won't launch it.  Even, though they implemented the pull policy of always update, supposedly to always have the most recent image.   Hence, I am running Watchtower, which actually pulls and retags the image, but only after the container has been launched.  Could it have a flag to search for images, then pull them and retag them?  Pretty simple concept, I think but tougher than it looks to get done.
